### ISSUE TITLE
error when no gradient function is provided

### ISSUE THREAD

**Issue by gica3618** [JohannesBuchner/PinNUTS#1]
When a sampler is initialised without gradient function (that is, `grad_logLikelihood=None`), and error is thrown by the helpers.py:
`NameError: name 'numerical_grad' is not defined`
Indeed, `numerical_grad` on line 53 of helpers.py is not defined. So I believe that currently, the code does not work if no gradient function is provided.
As an example to reproduce the error, one can replace `grad_logLikelihood` by `None` on line 185 of imf_examples.py.

**Comment by JohannesBuchner** [comment]
yeah, grad_logLikelihood=None is not intended to be supported here, but was supported in the original code. 

This is an odd error to receive, and a byproduct of stale code that should not be reached. Probably best would be to assert grad_logLikelihood is not None somewhere.

PRs are welcome.
